doc/source/development.rst

```diff
@@ -54,7 +54,15 @@ helpful.
    something like ``flake8 ray/python/ray/worker.py``. You may need to first run
    ``pip install flake8``.
 
-5. **Inspecting Redis shards by hand:** To inspect the primary Redis shard by
+5. **Autoformatting code**. We use ``yapf``
+   https://github.com/google/yapf for linting, and the config file is
+   located at ``.style.yapf``. We recommend adding ``.travis/yapf.sh``
+   to git's ``pre-push`` hook, so that you never have to worry about
```

Comment:
I'm curious why this is so slow, we only have ~90k lines of Python (even including rllib / dataframes), so it seems weird that it takes almost a minute to format that. I don't think we can recommend this unless we get formatting time down to some thing more reasonable, e.g. 5-10 seconds.

Reply:
Takes about 20 seconds on my MacBook

Reply:
Even a minute isn't that bad for pre push since you push far less than you commit

Reply:
It must depend on your hardware then, on my laptop it takes twice a long (4x i7 cores). Still, 20s is an eternity when pushing typically takes just a couple seconds.

Reply:
The only one that's faster is [`black`](https://github.com/ambv/black), since it checks if the files on disk have been changed. 

I would be perfectly happy to switch to it, since it offers no configuration at all, which makes debates about style moot, except I doubt the rest of the team would, given the lack of said configuration.

Reply:
FWIW, I've always been doing `git status` and then passing all changed files into

```
yapf --style=.style.yapf -i [all changed files]
```